goSrc/src/vmCaller/iroha/commands.go

```diff
@@ -0,0 +1,117 @@
+package iroha
+
+// #cgo CFLAGS: -I ../../../../irohad
+// #cgo LDFLAGS: -Wl,-unresolved-symbols=ignore-all
+// #include "ametsuchi/impl/proto_command_executor.h"
+// #include "ametsuchi/impl/proto_specific_query_executor.h"
+import "C"
+import (
+	"fmt"
+	"time"
+	"unsafe"
+
+	pb "iroha.protocol"
+
+	"github.com/golang/protobuf/proto"
+)
+
+var (
+	IrohaCommandExecutor unsafe.Pointer
+	IrohaQueryExecutor   unsafe.Pointer
+	Caller               string
+)
+
+// -----------------------Iroha commands---------------------------------------
+
+/*
+	Transfer assets between accounts
+*/
+func TransferAsset(src, dst, asset, amount string) error {
+	command := &pb.Command{Command: &pb.Command_TransferAsset{
+		TransferAsset: &pb.TransferAsset{
+			SrcAccountId:  src,
+			DestAccountId: dst,
+			AssetId:       asset,
+			Description:   "EVM asset transfer",
+			Amount:        amount,
+		}}}
+	commandResult, err := makeProtobufCmdAndExecute(IrohaCommandExecutor, command)
+	if err != nil {
+		return err
+	}
+	if commandResult.error_code != 0 {
+		error_extra := ""
+		error_extra_ptr := commandResult.error_extra.toStringAndRelease()
+		if error_extra_ptr != nil {
+			error_extra = ": " + *error_extra_ptr
+		}
+		return fmt.Errorf("[api.TransferIrohaAsset] error transferring asset nominated in %s from %s to %s%s", asset, src, dst, error_extra)
+	}
+
+	return nil
+}
+
+// -----------------------Iroha queries---------------------------------------
+
+// Queries asset balance of an account
+func GetAccountAssets(accountID string) ([]*pb.AccountAsset, error) {
+	query := &pb.Query{Payload: &pb.Query_Payload{
+		Meta: &pb.QueryPayloadMeta{
+			CreatedTime:      uint64(time.Now().UnixNano() / int64(time.Millisecond)),
+			CreatorAccountId: Caller,
+			QueryCounter:     1},
+		Query: &pb.Query_Payload_GetAccountAssets{
+			GetAccountAssets: &pb.GetAccountAssets{AccountId: accountID}}}}
+	queryResponse, err := makeProtobufQueryAndExecute(IrohaQueryExecutor, query)
+	if err != nil {
+		return []*pb.AccountAsset{}, err
+	}
+	switch response := queryResponse.Response.(type) {
+	case *pb.QueryResponse_ErrorResponse:
+		if response.ErrorResponse.Reason == pb.ErrorResponse_NO_ACCOUNT {
+			// No errors, but requested account does not exist
+			return []*pb.AccountAsset{}, nil
```

Comment:
Will this result in "0" returned in native getAssetBalance contract for non-existing account?

Reply:
not anymore